ONSdigital/ras-collection-instrument
# TBL 037 test suite for ras-collection-instrument

### What is the context of this PR?
To implement a consistent structure for testing. This includes a separate testing folder, scripts to run the tests and flake8, which also links through to the travis.yml. Coverage has also been added so the developer can call it from the console (coverage html) A virtual postges database has been added which will be called when tests are run. A limit set of tests have been added due to new development, so they are more examples of how the system works

### How to review 
1. Make sure ras-compose still works (./run.sh)
2. Update requirements file (pip install -r requirements.txt and pip install -r requirements_for_test.txt)
3. Run the test scripts eg ./script/run_test.sh and make sure everything passes
4. Run "Coverage html" from the command line and make sure a fold called hmtlcov is generated, open the index.html in a browser and see the results.

N.B This is not a full set of tests and just a sample due to development changing the end points.  The behave tests still need work from @liamtoozer however they were added to this branch
